// apollo-server-cache-memcached/dist/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MemcachedCache = void 0;
const memcached_1 = __importDefault(require("memcached"));
const util_1 = require("util");
class MemcachedCache {
    constructor(serverLocation, options) {
        this.defaultSetOptions = {
            ttl: 300,
        };
        const client = new memcached_1.default(serverLocation, options);
        client.del = (0, util_1.promisify)(client.del).bind(client);
        client.get = (0, util_1.promisify)(client.get).bind(client);
        client.set = (0, util_1.promisify)(client.set).bind(client);
        client.flush = (0, util_1.promisify)(client.flush).bind(client);
        this.client = client;
    }
    async set(key, value, options) {
        const { ttl } = Object.assign({}, this.defaultSetOptions, options);
        if (typeof ttl === 'number') {
            await this.client.set(key, value, ttl);
        }
        else {
            await this.client.set(key, value, 0);
        }
    }
    async get(key) {
        return await this.client.get(key);
    }
    async delete(key) {
        return await this.client.del(key);
    }
    async flush() {
        await this.client.flush();
    }
    async close() {
        this.client.end();
    }
}
exports.MemcachedCache = MemcachedCache;
